platform/viewer/cypress/integration/common/OHIFCornerstoneToolbar.spec.js

```diff
@@ -0,0 +1,509 @@
+            .should('have.attr', 'style', 'height: 33.3333%; width: 100%;')//checking if the layout is correct
+            .its('length')
+            .should('be.eq', 3);
+        })
+
+        //verify if layout has changed to 4 viewports 
+        cy.get('@layoutBtn').click();
+        cy.get('tbody > :nth-child(2) > :nth-child(2)').click();
+        cy.get('.viewport-container').then(($viewport) =>{
+          cy.wrap($viewport)
+            .should('have.attr', 'style', 'height: 50%; width: 50%;')//checking if the layout is correct
+            .its('length')
+            .should('be.eq', 4);
+        })
+
+        //verify if layout has changed to 6 viewports 
+        cy.get('@layoutBtn').click();
+        cy.get('tbody > :nth-child(2) > :nth-child(3)').click();
+        cy.get('.viewport-container').then(($viewport) =>{
+          cy.wrap($viewport)
+            .should('have.attr', 'style', 'height: 50%; width: 33.3333%;')//checking if the layout is correct
+            .its('length')
+            .should('be.eq', 6);
+        })
+
+        cy.get('@layoutBtn').click();
+        cy.get('tbody > :nth-child(3) > :nth-child(2)').click();
+        cy.get('.viewport-container').then(($viewport) =>{
+          cy.wrap($viewport)
+            .should('have.attr', 'style', 'height: 33.3333%; width: 50%;')//checking if the layout is correct
+            .its('length')
+            .should('be.eq', 6);
+        })
+
+        //verify if layout has changed to 9 viewports 
+        cy.get('@layoutBtn').click();
+        cy.get('tbody > :nth-child(3) > :nth-child(3)').click();
+        cy.get('.viewport-container').then(($viewport) =>{
+          cy.wrap($viewport)
+            .should('have.attr', 'style', 'height: 33.3333%; width: 33.3333%;')//checking if the layout is correct
+            .its('length')
+            .should('be.eq', 9);
+        })
+
+         //verify if layout has changed to 1 viewport 
+         cy.get('@layoutBtn').click();
+         cy.get('tbody > :nth-child(1) > :nth-child(1)').click();
+         cy.get('.viewport-container').then(($viewport) =>{
+           cy.wrap($viewport)
+             .should('have.attr', 'style', 'height: 100%; width: 100%;')//checking if the layout is correct
+             .its('length')
+             .should('be.eq', 1);
+         })
+    });
+
+
+    it('check if Invert tool will change the colors of the image in the viewport', () => {
+      //Click on More button
+      cy.get('.expandableToolMenu').click();
+      //Verify if overlay is displayed
+      cy.get('.tooltip-toolbar-overlay')
+        .should('be.visible');
+
+      //Click on Invert button
+      cy.get('.tooltip-inner > :nth-child(6)')
+        .click();
+
+      //check on cornerstone if image was inverted
+      cy.window()
+        .its('cornerstone')
+        .then(($cornerstone) => {
+          expect($cornerstone.getEnabledElements()[0].viewport.invert).to.be.true
+        });
+
+      cy.resetViewport();
+    });
```

Comment:
This, similarly, falls under "visual testing". Worth giving this a read: https://docs.cypress.io/guides/tooling/visual-testing.html#Visual-diff-elements

Something like this: https://github.com/meinaart/cypress-plugin-snapshots
Or this: https://docs.percy.io/docs/cypress

Would be better suited for a test like this.
I think it's best we remember these tests, but earmark them as potential future visual tests.

Reply:
Great observation. I'll keep this in mind and will implement some visual testing conditions on our next tests.